mozilla-ai/lumigator
# Enable usage inside GitHub Codespaces

## What's changing

This P.R. adds a new file `.devcontainer/codespaces/devcontainer.json` that allows to run and test Luminator inside [GitHub Codespaces](https://github.com/features/codespaces)

## How to test it

Create a codespace from the GitHub UI using `New with options`:

![Screenshot 2024-11-13 113320](https://github.com/user-attachments/assets/866f76ea-b068-4be4-bd5f-765c4b4c61af)

Make sure to select this branch and the `Lumigator`  config:

![image](https://github.com/user-attachments/assets/7a333078-3f27-490f-8a5d-18e66a4bd413)

## Additional notes for reviewers

By default GitHub uses `.devcontainer/devcontainer.json` when creating a codespace. Since the existing one is not meant to be used in codespaces, perhaps it could be moved to a subfolder like `.devcontainer/local/devcontainer.json`?


## I already...

Went through the full quickstart inside a codespace:

![Screenshot 2024-11-13 113024](https://github.com/user-attachments/assets/fc99910c-636b-4d27-8b7f-f2f2e21a5ff2)
![Screenshot 2024-11-13 113036](https://github.com/user-attachments/assets/0857c6c3-3861-4cf3-8ce9-21ad1e5eddb2)
